src/dataflow/src/render/join/linear_join.rs

```diff
@@ -164,138 +167,51 @@ where
     Trace(ArrangementImport<G, Row, T>),
 }
 
-impl<G, T> Context<G, MirRelationExpr, Row, T>
+impl<G, T> Context<G, Row, T>
 where
     G: Scope,
     G::Timestamp: Lattice + Refines<T>,
     T: Timestamp + Lattice,
 {
     pub fn render_join(
         &mut self,
-        relation_expr: &MirRelationExpr,
-        map_filter_project: MapFilterProject,
-        // TODO(frank): use this argument to create a region surrounding the join.
```

Comment:
I think the TODO is still legit. IMO we should put linear joins within a region.

Reply:
It is still legit, but there is an issue filed for putting regions around various subgraphs, linear joins included.

Reply:
Just double checked: https://github.com/MaterializeInc/materialize/issues/6880 for reference.